src/functions.php

```diff
@@ -113,6 +113,119 @@ function stream_for($resource = '', array $options = [])
     throw new \InvalidArgumentException('Invalid resource type: ' . gettype($resource));
 }
 
+
+/**
+ * Return a ServerRequestInterface populated with superglobals :
+ * $_GET
+ * $_POST
+ * $_COOKIE
+ * $_FILES
+ * $_SERVER
+ *
+ * @return ServerRequest
+ */
+function server_request_from_global()
+{
+    $method = !isset($_SERVER['REQUEST_METHOD']) ? 'GET' : $_SERVER['REQUEST_METHOD'];
+    $headers = [];
+    if (function_exists('apache_request_headers')) {
+        $headers = apache_request_headers();
+    }
+    $serverRequest = new ServerRequest(
+        $method,
+        uri_from_global(),
+        $headers,
+        stream_for(fopen('php://input', 'r+')),
+        $_SERVER);
+    $serverRequest = $serverRequest
+        ->withCookieParams($_COOKIE)
+        ->withQueryParams($_GET)
+        ->withParsedBody($_POST)
+        ->withUploadedFiles(uploaded_files_from_global());
+    return $serverRequest;
+}
+
+/**
+ * Return a UriInterface populated with data contained in superglobal $_SERVER
+ *
+ * @return Uri
+ */
+function uri_from_global()
```

Comment:
I think this would be better as a static factory method on the Uri class (e.g., `fromGlobals`).